ffi/diplomat/src/timezone.rs

```diff
@@ -137,6 +137,24 @@ pub mod ffi {
             Ok(())
         }
 
+        /// Sets the `time_zone_id` field from an IANA string by looking up
+        /// the corresponding BCP-47 string.
+        ///
+        /// Errors if the string is not a valid BCP-47 time zone ID.
+        #[diplomat::rust_link(icu::timezone::IanaToBcp47MapperBorrowed::get, FnInStruct)]
+        pub fn try_set_iana_time_zone_id(
+            &mut self,
+            mapper: &crate::iana_bcp47_mapper::ffi::ICU4XIanaToBcp47Mapper,
+            id: &str,
+        ) -> Result<(), ICU4XError> {
+            self.0.time_zone_id = mapper.0.as_borrowed().get(id);
```

Comment:
this overrides the current time zone if there's an error

Reply:
good catch; fixed